src/main/java/entities/User.java

```diff
@@ -95,24 +96,54 @@ public void createListing() {
     public void addCart() {
 
     }
+
     public void removeCart() {
 
     }
+
+    /**
+     * Creates a new review with this User as the reviewer, and the specified reviewer and rating
+     * then adds it to the reviewed User's reviews
+     *
+     * @param reviewed the User being reviewed/the User whose reviews the new Review will be added to
+     * @param rating   the rating given to the User being reviewed
+     */
     public void writeReview(User reviewed, int rating) {
         new ReviewCreator().createReview(this, reviewed, rating);
 
     }
-    public void removeReview() {
 
+    /**
+     * Removes a review from this User's list of reviews
+     *
+     * @param toBeRemoved the Review to be removed from this User's reviews
+     */
+    public void removeReview(Review toBeRemoved) {
+        this.reviews.remove(toBeRemoved);
     }
 
-    //TODO: check if this is necessary/who gets to hold reviews
-    public void addReview(Review review){
+    /**
+     * Adds a review to this User's list of reviews
+     *
+     * @param review the review to be added to this User's reviews
+     */
+    public void addReview(Review review) {
         this.reviews.add(review);
     }
-//    public int calculateRating() {
-//
-//    }
+
+    /**
+     * calculates the average integer rating earned by this User
+     *
+     * @return the average rating of all this User's reviews
+     */
+    public int calculateRating() {
+        double rating = 0;
+        for (Review review : reviews) {
+            rating += review.getRating();
+        }
+        rating /= reviews.size();
+        return (int) Math.round(rating);
+    }
```

Comment:
i feel like a review could be a double, and we should allow decimal ratings. maybe round to a tenth, but not to a whole number. implementation detail tho, so i'm gonna approve for now!